### ISSUE TITLE
Ghost Surfaces - Redux

### ISSUE THREAD

**Issue by etmoonshade** [Sigma88/Sigma-Dimensions#96]
Version: 1.10.1 w/ SD 0.10.7 and Galileo88's RESCALE mod at 10.625x, clean install. But see below...

This seems to be a popular theme with the open issues, no?

I'm seeing the same thing as #77 and #95 - a ghostly surface that only appears when you zoom out enough. If I try to land on where the surface _should_ be,  I end up exploding as I pass through terrain that isn't there. Interestingly, MechJeb seems to know where the surface is supposed to be (i.e. the true altitude reads roughly what I'd expect it to be, and goes negative as I sink into the terrain. At ground level, KSC appears to be floating above the ocean - presumably ~328m above it.

Now, I **KNEW** I was using the SD + RESCALE mods on KSP1.10.0 with an accidentally de-nyancatted version of Kopernicus for 1.9.1 (1.9.1 release 3, assuming it hasn't been fixed,) so I dug around in my old folders and found it. Indeed, it looks like something changed between KSP 1.10.0 and 1.10.1 to cause these issues.

I'll mention this to @R-T-B (who I guess I can't actually @, darn it,) but I figured I'd also drop you a bug report just in case.

Obviously I'm stacking old versions of mods on old versions of mods on new versions of KSP here, so I don't expect a whole bunch of support - I feel like I make a decent case that something went weird between the versions, at least.

**Comment by R-T-B** [comment]
I got the tag, lol.  No idea why this is happening unfortunately, this isn't really my mod though so I'm guessing something in sigma dimensions but am open to being show it's Kopernicus's fault as well.  At this point I have no real idea where to begin, though.

**Comment by etmoonshade** [comment]
Weird - it wouldn't auto-complete your name for whatever reason.

I figured one of y'all would have half a clue what's going on, at least. I saw a bunch of mentions of PQSes in the config files and previous troubleshooting steps, and I know there's been issues with that in Kopernicus during the BE testing (though they're supposedly fixed now?)

To provide a bit more clarity, I've attached some pictures:
![screenshot36](https://user-images.githubusercontent.com/279491/91651308-e21c2500-ea50-11ea-9605-0637d5e97a64.png)
![screenshot37](https://user-images.githubusercontent.com/279491/91651310-e34d5200-ea50-11ea-90fe-072cc9b9fb9d.png)
![screenshot38](https://user-images.githubusercontent.com/279491/91651312-e3e5e880-ea50-11ea-8ae4-ce234ad211e4.png)
![screenshot39](https://user-images.githubusercontent.com/279491/91651313-e47e7f00-ea50-11ea-97e5-b9d3174159b0.png)

Low, medium, and high-angle views from maximum zoom-out, and a view at ground level showing where the base is. I was going to try to get a landing shot as well, but I got too impatient trying to get away from the KSC with my terrible craft and blew it up when I hit the not-ground. :D

I'll also clear up my test cases a bit:
All are using the most recent version of SD at this time, from 20200229.
SD + Kopernicus 1.9.1-3 + KSP 1.10.0 = no issues
SD + Kopernicus 1.10.0-6 + KSP 1.10.0 = no issues
SD + Kopernicus 1.9.1-3 + KSP 1.10.1 = fail
SD + Kopernicus 1.10.1-6 + KSP 1.10.1 = fail
SD + Kopernicus 1.10.1-20 + KSP 1.10.1 = fail

I'm making the _assumption_ that it's some weird interaction of SD with 1.10.1 since @R-T-B  had so many problems with PQSes in Kopernicus (I think they were all in 1.10.1?) but I have nothing to support that specific analysis other than my test cases above.

**Comment by R-T-B** [comment]
The PQS issues Kopernicus experienced were fixed, yes, which is why I am suspect it could be our fault.  We went through that code with a fine toother comb, mostly.

Sounds like it could've happened in between 1.10.0 and 1.10.1, possibly.  Any chance of running the latest bleeding edge against 1.10.0 if I provide you with a special build?

**Comment by etmoonshade** [comment]
I mean, sure, I'll give it a shot if you want, but I'm pretty sure it's _not_ Kopernicus' fault here since I tried the 1.10.x-6 series on KSP 1.10.0 and 1.10.1, and it worked on one but not the other.

**Comment by etmoonshade** [comment]
@R-T-B provided me a build of 1.10.0-20, and I was not able to replicate the issue.

I've asked for a build of 1.10.1-20 with debug logging enabled - if I can get that, I'll provide the logs from testing on that, in case it's helpful in figuring out the issue here.

**Comment by Sigma88** [comment]
hi there

> @R-T-B provided me a build of 1.10.0-20, and I was not able to replicate the issue.

If you were not able to replicate the issue does this mean that the issues are gone?

> https://user-images.githubusercontent.com/279491/91651310-e34d5200-ea50-11ea-90fe-072cc9b9fb9d.png

this looks like an issue with the scaledspace fading in too early. could be an issue linked to SD but if you don't see the issue on KSP 1.10.0 then it's probably more complicated than just that


> https://user-images.githubusercontent.com/279491/91651313-e47e7f00-ea50-11ea-97e5-b9d3174159b0.png

this looks like it is related to #95 the solution should be the same (see https://github.com/Sigma88/Sigma-Dimensions/issues/81#issuecomment-683457530)

**Comment by etmoonshade** [comment]
Nah, sorry, I should have been more clear - the issue _does_ exist in 1.10.1, but not in 1.10.0, with (essentially?) the same code running (I'm not sure what magic @R-T-B is using to target multiple versions with a single codebase.)

I'll give that patch in the #81 comment a try - I already tried removing the lines you mentioned on #95 on my main save and it didn't work, though I did _not_ try it on a clean install.

**Comment by etmoonshade** [comment]
@Sigma88 - okay, I removed lines 11 and 12 from \GameData\Sigma\Dimensions\Configs\ReDimension\ResizePQSMods.cfg per #95, and I've added the following patch per #81 and your suggestion:
`@Kopernicus:FOR[SigDim2]
{
    @Body:HAS[@PQS]
	{
		@PQS
		{
			%minLevel = 1
			%maxLevel = 4
		}
	}
}
`

Still seeing the issue on 1.10.1.

**Comment by R-T-B** [comment]
Yeah 1.10.1 introduced...  something that breaks this.  Still not sure entirely what.  The "common code base magic" is really just a matter of building the code against different files.  The code itself is pretty version agnostic now as long as there aren't major changes (and of course, that's never a promise).  I no longer rely strictly on exact texture names, for example, and things like that.  That's all the magic really is.

**Comment by Sigma88** [comment]
> Still seeing the issue on 1.10.1.

Just to be clear, are you experiencing an issue that when you try to land your ship goes through the ground and explodes?

or is it just the ghost surfaces issue ?

**Comment by etmoonshade** [comment]
I'm treating the ghost surface thing as the primary issue when I'm reproducing it - most of the time, I'm not even attempting to land anywhere.

If I'm understanding how it works, it's not just a _visual_ bug, right? I'm getting slightly out of my depth here, but the "PQS" is basically the high-detail surface (whether it's land, structures, etc.,) which includes collision meshes, and that's what isn't appearing at surface level?

While I grab that log for @R-T-B , I can try a landing on something that actually exists as well and report back on it with an edit here.

**Comment by Sigma88** [comment]
> If I'm understanding how it works, it's not just a visual bug, right? I'm getting slightly out of my depth here, but the "PQS" is basically the high-detail surface (whether it's land, structures, etc.,) which includes collision meshes, and that's what isn't appearing at surface level?

yes, however I am not sure if your issue is caused by the lack of PQS or by the fact that it is just being hidden by the scaled space model (which is just an "image" of the planet, basically whatever you see in the tracking station or while far aways from the planet)

the issue of falling through seemed to be caused by a quirk in the new KSP that doesn't like to have non-integer PQS min/max-level

the patch you mentioned before should avoid this issue, while the "ghost" issue will not be solved by doing that

**Comment by etmoonshade** [comment]
> yes, however I am not sure if your issue is caused by the lack of PQS or by the fact that it is just being hidden by the scaled space model (which is just an "image" of the planet, basically whatever you see in the tracking station or while far aways from the planet)

I think it might be working in the reverse direction that you're thinking maybe?

When zoomed as far out as possible, that's the "scaled space" model, correct? And then PQS is at close range?

I see what I expect to see when I zoom out - the scaled space model (texture?) of "Cape Kerbnaveral," the shores area surrounding the KSC, and then the icon for the KSC right on top of it. When zooming in, that goes missing and (seemingly) is _actually_ a couple hundred km to the north (it's hard to tell for sure because of the low angle.) I'll scout it out next time I hit a stopping point in my current game.

**Comment by R-T-B** [comment]
Thanks for the logs.  I had a rough day at work, so I'll probably look at them tomorrow, but I hope I can discover something interesting within.

**Comment by etmoonshade** [comment]
Okay, remember how I said I had a clean install?

One thing I've done out of habit for a LONG time is to copy my settings.cfg from install to install, because it's a pain in the ass setting up all those custom keybindings.

Well, I just _looked_ at settings.cfg, and noticed that there's a ton of random cruft from the various things I've installed in there.

I did a Very Definitely Clean Install(TM) and _suddenly_ I'm not having this issue anymore.

My apologies for sending y'all on this wild goose chase. I'm closing this issue.

**Comment by Sigma88** [comment]
do you have a copy of the old cfg that was causing issues ?

and a copy of the new one that fixed the issue as well?

thanks

**Comment by etmoonshade** [comment]
... apparently not.
I mean, I do, but it's not causing the issue when copied into a different install. It's the only thing I changed, so the alternative is... I dunno, dodgy SSD? I did end up replacing the thing recently. Something I changed and don't remember it? One of my install files is bad, and I grabbed from a different place this time?

Even putting the SD/RESCALE files into the install that was previously causing the problem no longer does.

The entries I'm talking about in settings.cfg are these:
```TERRAIN
{
	preset = RSS_High
	version = 1.10.0
	PRESET
	{
		name = Low
		PLANET
		{
			name = Kerbin
			minDistance = 4
			minSubdivision = 1
			maxSubdivision = 8
		}
		PLANET
		{
			name = KerbinOcean
			minDistance = 4
			minSubdivision = 1
			maxSubdivision = 7
		}
		PLANET
		{
			name = Mun
			minDistance = 4
			minSubdivision = 1
			maxSubdivision = 7
		}
```

I've got entries like this for basically every planet pack I've installed (and then subsequently uninstalled, because I eventually ended up going RSS for the "real" size.) With that said, apparently they're not causing a problem anymore. I did notice that the graphics setting slider had reverted to low (it used to be some weird AUTOLOC text, if I remember right,) and had a lot more stops than just low/default/high - although the setting doesn't appear to stick. Maybe it had a weird setting? Although again, I can't seem to actually reproduce the issue anymore with moving the slider around...

I'm chalking this up to a bad report - if I ever manage to reproduce the problem again though, I'll have a few more things to check and I'll attach anything relevant that I find.
